src/libcore/iter/traits.rs

```diff
@@ -473,15 +473,15 @@ impl<'a, I: DoubleEndedIterator + ?Sized> DoubleEndedIterator for &'a mut I {
 /// impl ExactSizeIterator for Counter {
 ///     // We already have the number of iterations, so we can use it directly.
 ///     fn len(&self) -> usize {
-///         self.count
+///         5 - self.count
```

Comment:
Won't this overflow when `self.count == 6`? Seems like `self.count` should only be incremented in the `next` function in the `Some` branch.